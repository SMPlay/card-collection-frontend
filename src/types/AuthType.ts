export interface AuthType<T> {
  handleSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
  handleChange: (e: React.ChangeEvent<any>) => void;
  handleBlur: (e: any) => void;
  values: T;
  status: string;
  errors: T;
  touched: T;
}
