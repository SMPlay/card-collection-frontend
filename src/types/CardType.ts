export interface CardType {
  id: string;
  name: string;
  imageUrl: string;
  role: string;
  rarity: string;
  need: number;
  have: number;
  number: number;
}