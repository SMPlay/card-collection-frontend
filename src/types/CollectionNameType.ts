export interface CollectionNameType {
  id: string;
  name: string;
}
