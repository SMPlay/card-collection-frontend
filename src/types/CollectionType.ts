export interface CollectionType {
  name: string;
  imageUrl: string;
  id: string;
  cardsCount: number;
  release: number;
}
