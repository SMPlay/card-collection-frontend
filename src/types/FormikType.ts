interface Status {
  loading: boolean;
  error: string;
}

export interface FormikType<T> {
  handleSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
  handleChange: (e: React.ChangeEvent<any>) => void;
  values: T;
  status: Status;
  errors: T;
}
