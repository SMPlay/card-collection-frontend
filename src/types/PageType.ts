export interface PageType {
  pageName: string;
  url: string;
}
