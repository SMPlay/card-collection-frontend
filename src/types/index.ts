export { CollectionType } from './CollectionType'
export { PageType } from './PageType'