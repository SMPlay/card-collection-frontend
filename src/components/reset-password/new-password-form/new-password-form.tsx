import React from "react";
import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";
import { Box, Button, TextField, Typography } from "@material-ui/core";

import { FormikType } from "../../../types/FormikType";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
      "& .MuiTextField-root": {
        margin: theme.spacing(1),
        width: "60ch",
      },
    },
    newPasswordField: {
      display: "flex",
      flexDirection: "column",
    },
  })
);

export interface Values {
  newPassword?: string;
  confirmNewPassword?: string;
}

export const NewPasswordForm: React.FC<FormikType<Values>> = ({
  handleSubmit,
  handleChange,
  values,
  status,
  errors,
}) => {
  const styles = useStyles();

  return (
    <form onSubmit={handleSubmit} className={styles.root}>
      {status?.error ? (
        <Typography variant="h4" component="h2">
          {status?.error}
        </Typography>
      ) : (
        ""
      )}
      <Box m={2} className={styles.newPasswordField}>
        <TextField
          id="newPassword"
          value={values.newPassword}
          onChange={handleChange}
          label="Новый пароль"
          placeholder="Введите новый пароль"
          variant="outlined"
          type="password"
          error={errors.newPassword ? true : false}
          helperText={errors.newPassword}
          required
          disabled={status?.loading}
        />
        <TextField
          id="confirmNewPassword"
          value={values.confirmNewPassword}
          onChange={handleChange}
          label="Пароль ещё раз"
          placeholder="Введите пароль снова"
          variant="outlined"
          type="password"
          error={errors.confirmNewPassword ? true : false}
          helperText={errors.confirmNewPassword}
          required
          disabled={status?.loading}
        />
      </Box>
      <Button
        variant="contained"
        color="primary"
        type="submit"
        disabled={status?.loading}
      >
        Установить новый пароль
      </Button>
    </form>
  );
};
