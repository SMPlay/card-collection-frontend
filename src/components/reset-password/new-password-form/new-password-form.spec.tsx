import React from "react";
import { create } from "react-test-renderer";

import { NewPasswordForm, Values } from "./new-password-form";
import { FormikType } from "../../../types/FormikType";

describe("NewPasswordForm component", () => {
  let props: FormikType<Values>;

  beforeEach(() => {
    props = {
      handleSubmit: jest.fn(),
      handleChange: jest.fn(),
      values: {
        newPassword: "test",
        confirmNewPassword: "test"
      },
      status: {
        loading: false,
        error: "",
      },
      errors: {
        newPassword: "",
        confirmNewPassword: ""
      }
    }
  });

  it("should render", () => {
    const tree = create(<NewPasswordForm {...props}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
