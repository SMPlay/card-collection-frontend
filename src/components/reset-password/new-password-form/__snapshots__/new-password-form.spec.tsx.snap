// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPasswordForm component should render 1`] = `
<form
  className="makeStyles-root-1"
  onSubmit={[MockFunction]}
>
  
  <div
    className="MuiBox-root MuiBox-root-3 makeStyles-newPasswordField-2"
  >
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
        data-shrink={true}
        htmlFor="newPassword"
        id="newPassword-label"
      >
        Новый пароль
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="newPassword"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Введите новый пароль"
          required={true}
          type="password"
          value="test"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-6 PrivateNotchedOutline-legendNotched-7"
          >
            <span>
              Новый пароль
               *
            </span>
          </legend>
        </fieldset>
      </div>
      
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
        data-shrink={true}
        htmlFor="confirmNewPassword"
        id="confirmNewPassword-label"
      >
        Пароль ещё раз
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="confirmNewPassword"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Введите пароль снова"
          required={true}
          type="password"
          value="test"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-6 PrivateNotchedOutline-legendNotched-7"
          >
            <span>
              Пароль ещё раз
               *
            </span>
          </legend>
        </fieldset>
      </div>
      
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    <span
      className="MuiButton-label"
    >
      Установить новый пароль
    </span>
  </button>
</form>
`;
